### DISCUSSION TITLE
Any example code of using sdbus-cpp with BlueZ for a BLE central role (scanning, reading/writing characteristics)?

### DISCUSSION THREAD

**Discussion by brouhaha** [Kistler-Group/sdbus-cpp#308] [General]
I'm trying to use sdbus-cpp to implement a BLE central role, to scan for a peripheral, open a connection, and read and write characteristics. I have working Python code from "The Bluetooth Technology for Linux Developers Study Guide", which uses Dbus from Python, but I'm having a hard time understanding how to do the equivalent things in C++ using sdbus-cpp, even looking at the sdbus-cpp documentation and example. Does anyone have any example BLE central code they'd be willing to share?

I don't particularly need BLE peripheral side code, as I'm using bluez-dbus-cpp for that (which uses sdbus-cpp).

**Comment by sangelovic** [comment]
Hi, I am not familiar with BLE, "The Bluetooth Technology for Linux Developers Study Guide" nor bluez-dbus-cpp implementation, so I'm afraid I can't help you with this one. I can only answer sdbus-c++-specific questions related to its design, API and its usage.
..

**Comment by parsley72** [comment]
I started with [Me and the Devil BlueZ: Reading BLE Sensors From C++](https://www.mongodb.com/developer/languages/cpp/me-and-the-devil-bluez-2)

**Comment by jwinarske** [comment]
@brouhaha You might take a look at https://github.com/weareaudiofile/bluez-dbus-cpp

**Comment by paulocoutinhox** [comment]
If you can, help us implement here: https://github.com/bitchatz/bitchatz-cpp/pull/3
